pkg/multicluster/register/interfaces.go

```diff
@@ -45,11 +46,25 @@ type Options struct {
 	// Defaults to 'cluster.local'
 	// Read more: https://kubernetes.io/docs/tasks/administer-cluster/dns-custom-nameservers/
 	ClusterDomain string
+
+	RegistrationMetadata RegistrationMetadata
+
+	RbacOptions RbacOptions
 }
 
-type RbacOptions struct {
-	Options
+// Optional additional metadata to persist to registration output resources.
+type RegistrationMetadata struct {
+	// Metadata about the provider for cloud hosted k8s clusters.
+	ProviderInfo *v1alpha1.KubernetesClusterSpec_ProviderInfo
 
+	// Labels to add to registration output resources (KubernetesCluster and Secret).
+	ResourceLabels map[string]string
+
+	// The set of PolicyRules for the cluster roles created on the remote cluster upon registration.
+	ClusterRolePolicyRules []*v1alpha1.PolicyRule
```

Comment:
are they always cluster roles? this wasn't clear to me

Reply:
do you think it makes sense to put non-cluster rbac here?

Reply:
It looks like no project is registering with Roles, so I opted not to include that here

Reply:
Discussed offline, there's reason to include this logic (we anticipate registering with Roles to become a requested use case in the coming months)